src/Console/CommandLoader.php

```diff
@@ -0,0 +1,130 @@
+
+use Redaxo\Core\Console\Command\AddonActivateCommand;
+use Redaxo\Core\Console\Command\AddonDeactivateCommand;
+use Redaxo\Core\Console\Command\AddonDeleteCommand;
+use Redaxo\Core\Console\Command\AddonInstallCommand;
+use Redaxo\Core\Console\Command\AddonListCommand;
+use Redaxo\Core\Console\Command\AddonRunUpdateCommand;
+use Redaxo\Core\Console\Command\AddonUninstallCommand;
+use Redaxo\Core\Console\Command\AssetsCompileStylesCommand;
+use Redaxo\Core\Console\Command\AssetsSyncCommand;
+use Redaxo\Core\Console\Command\CacheClearCommand;
+use Redaxo\Core\Console\Command\CommandAbstract;
+use Redaxo\Core\Console\Command\ConfigGetCommand;
+use Redaxo\Core\Console\Command\ConfigSetCommand;
+use Redaxo\Core\Console\Command\CronjobRunCommand;
+use Redaxo\Core\Console\Command\DatabaseConnectionOptionsCommand;
+use Redaxo\Core\Console\Command\DatabaseDumpSchemaCommand;
+use Redaxo\Core\Console\Command\DatabaseSetConnectionCommand;
+use Redaxo\Core\Console\Command\SetupCheckCommand;
+use Redaxo\Core\Console\Command\SetupRunCommand;
+use Redaxo\Core\Console\Command\SystemReportCommand;
+use Redaxo\Core\Console\Command\UserCreateCommand;
+use Redaxo\Core\Console\Command\UserSetPasswordCommand;
+use Redaxo\Core\Core;
+use rex_addon;
+use rex_exception;
+use Symfony\Component\Console\CommandLoader\CommandLoaderInterface;
+use Symfony\Component\Console\Exception\CommandNotFoundException;
+
+use function gettype;
+use function is_array;
+
+/**
+ * @internal
+ */
+class CommandLoader implements CommandLoaderInterface
+{
+    /** @var array<string, array{class: class-string<CommandAbstract>, package?: rex_addon}> */
+    private $commands = [];
+
+    public function __construct()
+    {
+        $commands = [
+            'cache:clear' => CacheClearCommand::class,
+            'config:get' => ConfigGetCommand::class,
+            'config:set' => ConfigSetCommand::class,
+            'db:connection-options' => DatabaseConnectionOptionsCommand::class,
+            'db:set-connection' => DatabaseSetConnectionCommand::class,
+            'setup:check' => SetupCheckCommand::class,
+            'setup:run' => SetupRunCommand::class,
+            'user:create' => UserCreateCommand::class,
+        ];
+
+        if (!Core::isSetup()) {
+            $commands = array_merge($commands, [
+                'assets:sync' => AssetsSyncCommand::class,
+                'assets:compile-styles' => AssetsCompileStylesCommand::class,
+                'cronjob:run' => CronjobRunCommand::class,
+                'db:dump-schema' => DatabaseDumpSchemaCommand::class,
+                'addon:activate' => AddonActivateCommand::class,
+                'addon:deactivate' => AddonDeactivateCommand::class,
+                'addon:delete' => AddonDeleteCommand::class,
+                'addon:list' => AddonListCommand::class,
+                'addon:install' => AddonInstallCommand::class,
+                'addon:run-update-script' => AddonRunUpdateCommand::class,
```

Comment:
Würde hier `Script` ruhig auch in den Command-Klassennamen mit aufnehmen, auch wenn es recht lang ist.